### PR TITLE
Version 0.19.0

### PR DESCRIPTION
**Pull request by Kludex**
Changes: https://github.com/encode/starlette/compare/0.18.0...master
Draft GitHub release: https://github.com/encode/starlette/releases

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## 0.19.0
+
+January, 27, 2022
+
+#### Added
+
+* Add `headers` parameter to `HTTPException` [#1435]().
+* Responses with `405 Method Not Allowed` returned from Starlette now insert an `Allow` header, as described by [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.5) [#1436]().
+* The `content` argument in `JSONResponse` is now required [#1431]().
```
Comment (Kludex) [inline docs/release-notes.md]: This is the breaking change.

File: docs/release-notes.md [hunk 2]
```diff
@@ -1,3 +1,14 @@
+## 0.19.0
+
+January, 28, 2022
+
+#### Added
+
+* Add `headers` parameter to `HTTPException` [#1435](https://github.com/encode/starlette/pull/1435).
```
Comment (adriangb) [inline docs/release-notes.md:7]: FYI @tiangolo you should be able to clean this up in FastAPI after this release

Comment (Kludex) [reply docs/release-notes.md:7]: https://github.com/tiangolo/fastapi/pull/4488 :+1:

File: docs/release-notes.md [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+* Internal responses with `405` status code insert an `Allow` header, as described by [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.5) [#1436](https://github.com/encode/starlette/pull/1436).
+* The `content` argument in `JSONResponse` is now required [#1431](https://github.com/encode/starlette/pull/1431).
+* Add custom URL convertor register [#1437](https://github.com/encode/starlette/pull/1437).
```
Comment (Kludex) [inline docs/release-notes.md:10]:
```suggestion
* Add custom URL convertor register [#1437](https://github.com/encode/starlette/pull/1437).
* Add `raw_path` to `TestClient` scope [#1445](https://github.com/encode/starlette/pull/1445).
```

File: docs/release-notes.md [hunk 4]
```diff
@@ -1,8 +1,22 @@
+## 0.19.0
+
+January, 28, 2022
+
+### Added
+* Add `headers` parameter to `HTTPException` [#1435](https://github.com/encode/starlette/pull/1435).
+* Internal responses with `405` status code insert an `Allow` header, as described by [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.5) [#1436](https://github.com/encode/starlette/pull/1436).
+* The `content` argument in `JSONResponse` is now required [#1431](https://github.com/encode/starlette/pull/1431).
+* Add custom URL convertor register [#1437](https://github.com/encode/starlette/pull/1437).
+* Add `raw_path` to `TestClient` scope [#1445](https://github.com/encode/starlette/pull/1445).
```
Comment (Kludex) [inline docs/release-notes.md:13]:
```suggestion
* Add `raw_path` to `TestClient` scope [#1445](https://github.com/encode/starlette/pull/1445).
* Bump `anyio` minimal version to `3.4.0` [#1421](https://github.com/encode/starlette/pull/#1421).
```

File: docs/release-notes.md [hunk 5]
```diff
@@ -1,8 +1,23 @@
...
+* Bump `anyio` minimal version to `3.4.0` [#1421](https://github.com/encode/starlette/pull/#1421).
```
Comment (Kludex) [inline docs/release-notes.md]:
```suggestion
* Bump `anyio` minimal version to `3.4.0` [#1421](https://github.com/encode/starlette/pull/1421).
```

File: docs/release-notes.md [hunk 6]
```diff
@@ -1,8 +1,23 @@
...
+### Added
+* Add `headers` parameter to `HTTPException` [#1435](https://github.com/encode/starlette/pull/1435).
+* Internal responses with `405` status code insert an `Allow` header, as described by [RFC 7231](https://datatracker.ietf.org/doc/html/rfc7231#section-6.5.5) [#1436](https://github.com/encode/starlette/pull/1436).
+* The `content` argument in `JSONResponse` is now required [#1431](https://github.com/encode/starlette/pull/1431).
+* Add custom URL convertor register [#1437](https://github.com/encode/starlette/pull/1437).
+* Add `raw_path` to `TestClient` scope [#1445](https://github.com/encode/starlette/pull/1445).
+* Bump `anyio` minimal version to `3.4.0` [#1421](https://github.com/encode/starlette/pull/1421).
+
+### Fixed
+* Prevent `BaseHTTPMiddleware` from hiding errors of `StreamingResponse` and mounted applications [#1459](https://github.com/encode/starlette/pull/1459).
```
Comment (Kludex) [inline docs/release-notes.md:20]:
```suggestion
* Prevent `BaseHTTPMiddleware` from hiding errors of `StreamingResponse` and mounted applications [#1459](https://github.com/encode/starlette/pull/1459).
* `Request.client` is now compliant with the ASGI specifications [#1462](https://github.com/encode/starlette/pull/1462).
```

File: docs/release-notes.md [hunk 7]
```diff
@@ -1,8 +1,23 @@
+## 0.19.0
+
+January, 28, 2022
+
+### Added
```
Comment (Kludex) [inline docs/release-notes.md:5]:
```suggestion
### Added
* Error handler will always run, even if the error happens on a background task [#761](https://github.com/encode/starlette/pull/761).
```

File: docs/release-notes.md [hunk 8]
```diff
@@ -1,8 +1,23 @@
+## 0.19.0
+
+January, 28, 2022
```
Comment (Kludex) [inline docs/release-notes.md]:
```suggestion
February, 1, 2022
```

File: docs/release-notes.md [hunk 9]
```diff
@@ -1,8 +1,37 @@
+## 0.19.0
+
+March, 9, 2022
```
Comment (aminalaee) [inline docs/release-notes.md]: Sorry to be too picky but I think the pattern is `March 9, 2022`?

Comment (Kludex) [reply docs/release-notes.md]:
```suggestion
March 9, 2022
```

**Review by aminalaee — APPROVED** [review]
👍

**Pull request merged**